server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java

```diff
@@ -306,12 +306,12 @@ protected void createTemplateWithinZones(TemplateProfile profile, VMTemplateVO t
             zonesIds = _dcDao.listAllZones().stream().map(DataCenterVO::getId).collect(Collectors.toList());
         }
 
-        List<DataStore> imageStores = getImageStoresThrowsExceptionIfNotFound(zonesIds, profile);
 
         for (long zoneId : zonesIds) {
             DataStore imageStore = verifyHeuristicRulesForZone(template, zoneId);
 
             if (imageStore == null) {
+                List<DataStore> imageStores = getImageStoresThrowsExceptionIfNotFound(List.of(zoneId), profile);
```

Comment:
@weizhouapache, just one remark, with these changes the method `getImageStoresThrowsExceptionIfNotFound` does not need to receive a List, could you refactor to receive only the zone ID?

Reply:
@BryanMLima 
yes, thanks. PR is updated